### PR TITLE
async UTransport and L2 APIs

### PR DESCRIPTION
**Pull request by stevenhartley**
The following change makes the UTransport APIs return CompletionStage<Void> in lieu of UStatus so that the operations can be asynchronous. The methods then return Void if successful or completes with a failed UStatusException with the UStatus information.

#124

### CODE REVIEW

File: src/test/java/org/eclipse/uprotocol/communication/TestUTransport.java [hunk 1]
```diff
@@ -26,6 +33,24 @@ public class TestUTransport implements UTransport {
...
+                        .setTopic(subscriptionRequest.getTopic())
+                        .build();
+                    return UMessageBuilder.response(request.getAttributes()).build(UPayload.pack(subResponse));
+                } catch (InvalidProtocolBufferException e) {
+                    return UMessageBuilder.response(request.getAttributes()).build(
+                        UPayload.pack(UnsubscribeResponse.newBuilder().build()));
+                }
+            } else {
+                return UMessageBuilder.response(request.getAttributes()).build(
+                    UPayload.pack(UnsubscribeResponse.newBuilder().build()));
```
Comment (dddj698) [inline src/test/java/org/eclipse/uprotocol/communication/TestUTransport.java:51]: I think we should copy request id to response reqId here

File: src/test/java/org/eclipse/uprotocol/communication/InMemorySubscriberTest.java [hunk 2]
```diff
@@ -46,21 +49,37 @@ public void onReceive(UMessage message) {
...
+    public void test_subscribe_and_then_unsubscribe_happy_path() {
+        UUri topic = createTopic();
+        UTransport transport = new TestUTransport();
+        UListener listener = new UListener() {
+            @Override
+            public void onReceive(UMessage message) {
+                // Do nothing
+            }
+        };
+        Subscriber subscriber = new InMemorySubscriber(transport, new InMemoryRpcClient(transport));
```
Comment (dddj698) [inline src/test/java/org/eclipse/uprotocol/communication/InMemorySubscriberTest.java:63]: Probably not a good idea to test both ImMemoryRpcClient and InMemorySubscriber in the same unit test. Suggest to mock InMemoryRpcClient with mocking library like mockito

File: src/test/java/org/eclipse/uprotocol/communication/TestUTransport.java (hunk 1 above)
Comment (stevenhartley) [reply src/test/java/org/eclipse/uprotocol/communication/TestUTransport.java:51]: the request id is in the UAtributes that is fed to the UMessageBuilder.response() static method

Comment (stevenhartley) [reply src/test/java/org/eclipse/uprotocol/communication/TestUTransport.java:51]: The requestID is copied from the UMessageBuilder.response() static method when it is passed the request.getAttributes()

File: src/test/java/org/eclipse/uprotocol/communication/InMemorySubscriberTest.java (hunk 2 above)
Comment (stevenhartley) [reply src/test/java/org/eclipse/uprotocol/communication/InMemorySubscriberTest.java:63]: Good point. I will take this for the next PR. Can you point me to an example so I can see?

**Pull request merged**